apps/test-app/frontend/src/components/table-widget/ExperimentalTableWidget.tsx

```diff
@@ -0,0 +1,167 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+import { useCallback, useEffect, useRef, useState } from "react";
+import { PropertyRecord } from "@itwin/appui-abstract";
+import { IModelConnection } from "@itwin/core-frontend";
+import { ProgressRadial } from "@itwin/itwinui-react";
+import { TableCellRenderer, TableColumnDefinition, TableRowDefinition, usePresentationTableWithUnifiedSelection } from "@itwin/presentation-components";
+import { flexRender, getCoreRowModel, useReactTable } from "@tanstack/react-table";
+
+export interface TableWidgetProps {
+  imodel: IModelConnection;
+  rulesetId?: string;
+}
+
+export function ExperimentalTableWidget(props: TableWidgetProps) {
+  const { imodel, rulesetId } = props;
+
+  if (!rulesetId) {
+    return null;
+  }
+
+  return <PresentationTable imodel={imodel} rulesetId={rulesetId} />;
+}
+
+interface PresentationTableProps {
+  imodel: IModelConnection;
+  rulesetId: string;
+}
+
+function PresentationTable(props: PresentationTableProps) {
+  const { imodel, rulesetId } = props;
+
+  const { columns, rows, isLoading, loadMoreRows, selectedRows, onSelect } = usePresentationTableWithUnifiedSelection({
+    imodel,
+    ruleset: rulesetId,
+    pageSize: 20,
+    columnMapper: mapTableColumns,
+    rowMapper: mapTableRow,
+  });
+
+  const visibleColumns = columns?.slice(0, 5);
+
+  const [rowSelection, setRowSelection] = useState<Record<string, boolean>>({});
+
+  useEffect(() => {
+    // create row selection state based on rows selected in unified selection
+    const selection: Record<string, boolean> = {};
+    selectedRows.forEach((row) => {
+      selection[row.id] = true;
+    });
+    setRowSelection(selection);
+  }, [selectedRows]);
+
+  const table = useReactTable({
+    data: rows,
+    columns: visibleColumns ?? [],
+    state: {
+      rowSelection,
+    },
+    enableRowSelection: true,
+    onRowSelectionChange: (updater) => {
+      const newRowSelection = typeof updater === "function" ? updater(rowSelection) : updater;
+
+      // collect selected row ids
+      const newSelectedRows: string[] = [];
+      for (const rowId in newRowSelection) {
+        if (rowId in newRowSelection && newRowSelection[rowId]) {
+          newSelectedRows.push(rowId);
+        }
+      }
```

Comment:
I am going to trust your experience and change it to your suggestion😃

Reply:
From your response it sounds like and explanation on why I think my approach is more readable is necessary... 
1. Using `Object.entries` allows us to avoid the `if (rowId in newRowSelection)` check.
2. By naming value part of the entry in `filter` we provide meaning to it. Type of `newRowSelection` is unclear, which makes the `if (newRowSelection[rowId])` check very vague.

Reply:
Thanks, it is indeed more clear this way